src/expr.cpp

```diff
@@ -7358,6 +7436,26 @@ Expr *SizeOfExpr::Optimize() { return this; }
 
 int SizeOfExpr::EstimateCost() const { return 0; }
 
+llvm::Constant *SizeOfExpr::GetConstant(const Type *type) const {
```

Comment:
I'm not sure why, but this produces incorrect code:
`uniform int s = sizeof (varying int);`
It treats "varying int" as uniform and initializes s with 4.

Reply:
Also, there should be a warning or error, in case of assigning size of varying types to global variables when doing multi-target compilation with different data layouts.

Reply:
multi-target compilation causes some issues. I'll have to rework that a little.